metrotranscom/doorway
# feat: frontend map pins

This PR addresses [#923](https://github.com/metrotranscom/doorway/issues/923)


## Description

This PR updates the map experience to show all pins, not just the pins in the list.

To accommodate this ask and maintain UX, it also adds the following:
- Map clustering of pins
- Filtering the list view based on what is visible in the map
- A new mobile view that toggles between list and map vs the old mobile drawer
- A full screen experience of map navigating on both desktop and mobile

## How Can This Be Tested/Reviewed?

- Ensure you have the GOOGLE_MAPS_API_KEY and GOOGLE_MAPS_MAP_ID environment variables set
- Run the new `yarn setup:large` command
- Ensure that as you pan around the map and filter, that the map and list reflect the listings you would expect
- There is more context for other specific visual changes within comments on the PR!
- There is no design. This has been through a prelim design review w Em and will go through another after merge when design can play around w it.

## Author Checklist:


## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates
